flow/envs/base.py

```diff
@@ -148,6 +148,10 @@ def __init__(self,
         self.state = None
         self.obs_var_labels = []
 
+        # track IDs that have ever been observed in the system
```

Comment:
Can you also incorporate the changes you made in multiagent/base.py into the singleagent env for symmetry? More specifically, add logic that updates `self.observed_ids`

Reply:
Sure thing.